libgalois/include/katana/PropertyGraph.h

```diff
@@ -193,9 +193,11 @@ class KATANA_EXPORT PropertyGraph {
   Result<void> Validate();
 
   Result<void> DoWrite(
-      tsuba::RDGHandle handle, const std::string& command_line);
+      tsuba::RDGHandle handle, const std::string& command_line,
+      bool retain_version);
   Result<void> WriteGraph(
-      const std::string& uri, const std::string& command_line);
+      const std::string& uri, const std::string& command_line,
+      bool retain_graph);
 
```

Comment:
I do not like the name `retain_version` when it comes to writing.  My model is every write to the graph creates a new version.  Are you adding a `WriteView`?  A `WriteView` operation would not increase the graph's version number.

Reply:
This was the cleanest way to do it. Otherwise, we will have a DoWrite, DoWriteView and a WriteGraph and a WriteGraphView. I think its too much considering the only difference is in whether or not the version number is incremented.

Reply:
I disagree.  Boolean parameters are tough to use because you can't tell from the call site what they mean.  But fundamentally, a graph is read-only.  If you write the graph, the version number has to increment.  Creating a view is not writing a graph, it is changing the format of an existing graph.  `Write(false)` does not make that clear and actually hides what is going on.

I'm not worried about naming internal functions.  I suspect we will be able to merge them sensibly.

Reply:
Boolean is definitely not the most readable way to do it. I have added PropertyGraph::WriteView and added RDG::RDGVersioningPolicy which is used by DoWrite to dictate whether the RDG Version should change with the store. Thanks.

Reply:
Thank you!